### PR TITLE
fix(codegen-scala): correctly determine value entity descriptors to add

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ValueEntitySourceGeneratorSuite.scala [hunk 1]
```diff
@@ -140,7 +140,7 @@ class MyValueEntityProvider private(entityFactory: ValueEntityContext => MyValue
     new MyValueEntityHandler(entityFactory(context))
 
   override final val additionalDescriptors =
-    service.MyServiceProto.javaDescriptor :: Nil
+    DomainProto.javaDescriptor :: service.MyServiceProto.javaDescriptor :: com.external.ExternalDomainProto.javaDescriptor :: Nil
```
Comment (patriknw) [inline codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ValueEntitySourceGeneratorSuite.scala:143]: yeah, I had a hunch that something was missing there

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/FullyQualifiedNameExtractorSuite.scala [hunk 2]
```diff
@@ -61,6 +61,9 @@ class FullyQualifiedNameExtractorSuite extends munit.FunSuite {
     val fqn = fqnExtractor(stateCart)
     assertNoDiff(fqn.name, "Cart")
     assertNoDiff(fqn.parent.scalaPackage, "com.example.shoppingcart.domain.shoppingcart_domain")
+    assertNoDiff(
```
Comment (patriknw) [inline codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/FullyQualifiedNameExtractorSuite.scala:64]: I'm not sure what the difference of `assertNoDiff` and `assertEquals` is, but this looks like it should be an equals?

File: codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ViewServiceSourceGenerator.scala [hunk 3]
```diff
@@ -191,7 +191,7 @@ object ViewServiceSourceGenerator {
         |
         |  @Override
         |  public final Descriptors.ServiceDescriptor serviceDescriptor() {
-        |    return ${view.descriptorObject.name}.getDescriptor().findServiceByName("${view.fqn.protoName}");
+        |    return ${typeName(view.fqn.descriptorImport)}.getDescriptor().findServiceByName("${view.fqn.protoName}");
```
Comment (patriknw) [inline codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ViewServiceSourceGenerator.scala:194]: nice to use same in Java codegen

File: codegen/core/src/main/scala/com/lightbend/akkasls/codegen/Packaging.scala [hunk 4]
```diff
@@ -32,20 +39,40 @@ case class FullyQualifiedName(protoName: String, name: String, parent: PackageNa
...
   def fileBasename =
     parent.javaPackage.replace('.', '/') + "/" + name
+
 }
 
 object FullyQualifiedName {
-  def apply(name: String, parent: PackageNaming): FullyQualifiedName =
-    FullyQualifiedName(name, name, parent)
-
+  // FIXME should only be used for testing, move there
```
Comment (patriknw) [inline codegen/core/src/main/scala/com/lightbend/akkasls/codegen/Packaging.scala:54]: move to TestData?

**Review by patriknw — APPROVED** [review]
LGTM

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/FullyQualifiedNameExtractorSuite.scala (hunk 2 above)
Comment (raboof) [reply codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/FullyQualifiedNameExtractorSuite.scala:64]: https://scalameta.org/munit/docs/assertions.html#assertnodiff

> The difference between assertNoDiff() and assertEquals() is that assertEquals() fails according to the == method while assertNoDiff() ignores non-visible differences such as trailing/leading whitespace, Windows/Unix newlines and ANSI color codes

Hmm, maybe we want `assertEquals` always after all?

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ValueEntitySourceGeneratorSuite.scala (hunk 1 above)
Comment (RayRoestenburg) [reply codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ValueEntitySourceGeneratorSuite.scala:143]: aha!

**Comment by RayRoestenburg** [comment]
Small fix here: https://github.com/lightbend/akkaserverless-java-sdk/pull/482/commits/163c216d94b6cc228b25ddbec46f09e976ed8533

**Comment by RayRoestenburg** [comment]
I rebased https://github.com/lightbend/akkaserverless-java-sdk/pull/482 on top of this PR here, and added the fix there, all looks good.

**Comment by RayRoestenburg** [comment]
Pulled this into https://github.com/lightbend/akkaserverless-java-sdk/pull/482

**Pull request closed without merging**